SSINS/sky_subtract.py

```diff
@@ -114,28 +114,64 @@ def diff(self):
         attributes are also adjusted so that the resulting SS object passes
         UVData.check()
         """
-
+        # want to reorder to baseline order
+        # baseline order guarantees that the baseline time axis is oriented so that we can safely assume where indices for diffing are
+        # by looking at the blt axis array
         if self.blt_order != 'baseline':
             warnings.warn("Reordering data array to baseline order to perform differencing.")
             self.reorder_blts(order='baseline')
 
-        # index accumulator
-        ind_acc = 0
+        # basline index: indicates which bl (in bl order) is being processed
+        bl_num = 0
+
         # Just treat every baseline independently - make no assumptions other than what is guaranteed by baseline ordered objects
+
+        # the baseline array is the baseline for each member of the baseline-time array, so given it is in baseline form it will look like
+        # [ bl1, bl1, bl1, bl1 ... bl1, bl2, bl2, bl2, bl2 ... bl2, bl3 ... bl(n) ]
+        # so differencing it will give zero for most entries and some value where there is a boundary (which we set to 1)
+        testarray = (self.baseline_array[1:] - self.baseline_array[:-1])
+
+        bltaxisboundaries = np.nonzero(testarray)
+        bltaxisboundaries = np.append(bltaxisboundaries, np.size(testarray, 0)) # add last value to end of array (diff won't be able to catch this)
+        bltaxisboundaries = bltaxisboundaries + 1                               # offset entries to correct distance between 0th index and 1st
+        bltaxisboundaries = np.insert(bltaxisboundaries, 0, 0)                  # add 0 to beginning of array (diff won't be able to catch this)
+
+        #need a second array representing what the output's shape will look like -- it's smaller by Nblts because of diffing
+        diffed_baseline_array = self.baseline_array
+        for bl in np.unique(self.baseline_array):
+            index = (np.argwhere(diffed_baseline_array == bl)[0:1])
+            diffed_baseline_array = np.delete(diffed_baseline_array, index, 0)
+
+        testarray = (diffed_baseline_array[1:] - diffed_baseline_array[:-1])
+        bltaxisboundaries2 = np.nonzero(testarray)
+        bltaxisboundaries2 = np.append(bltaxisboundaries2, np.size(testarray, 0)) # add last value to end of array (diff won't be able to catch this)
+        bltaxisboundaries2 = bltaxisboundaries2 + 1                               # offset entries to correct distance between 0th index and 1st
+        bltaxisboundaries2 = np.insert(bltaxisboundaries2, 0, 0)                  # add 0 to beginning of array (diff won't be able to catch this)
+
+        # loop through each baseline in the baseline_array, noting that the baseline_array will have each bl repeated for ntimes
+        # (so we need unique)
         for bl in np.unique(self.baseline_array):
```

Comment:
`for bl_num, bl in enumerate(np.unique(self.baseline_array))` (check my syntax on this) makes it so you don't have to manually increment `bl_num`, lets the reader see what bl_num is up front instead of hunting for it above and below, and least importantly, is more pythonic. You could also move `bl_num` next to the `for` statement.